CONTRIBUTING.md

```diff
@@ -0,0 +1,217 @@
+# Contributing Guide
+
+- [Contributing Guide](#contributing-guide)
+  - [Participating in the Project](#participating-in-the-project)
+    - [Community Participant](#community-participant)
+    - [Contributor](#contributor)
+    - [Maintainer](#maintainer)
+  - [Ways to Contribute](#ways-to-contribute)
+  - [Find an Issue](#find-an-issue)
+  - [Ask for Help](#ask-for-help)
+  - [Pull Request Lifecycle](#pull-request-lifecycle)
+  - [Development Environment Setup](#development-environment-setup)
+    - [Build](#build)
+    - [E2E Testing](#e2e-testing)
+  - [Sign Your Commits](#sign-your-commits)
+    - [DCO](#dco)
+  - [Pull Request Checklist](#pull-request-checklist)
+    - [Build](#build-1)
+    - [Testing](#testing)
+      - [E2E Testing](#e2e-testing-1)
+
+Welcome! We are glad that you want to contribute to our project! 💖
+
+As you get started, you are in the best position to give us feedback on areas of
+our project that we need help with including:
+
+* Problems found during setting up a new developer environment
+* Bugs in our automation scripts
+
+If anything doesn't make sense, or doesn't work when you run it, please open a
+bug report and let us know!
+
+## Participating in the Project
+
+There are a number of ways to participate in this project. As the project evolves and grows, we will define a more formal governance model. For now, this document describes various ways community members might participate.
+
+### Community Participant
+
+A Community Participant engages with the project and its community, contributing their time, thoughts, etc. Community participants are usually users who have stopped being anonymous and started being active in project discussions.
+
+### Contributor
+
+A Contributor contributes directly to the project. Contributions need not be code. People at the Contributor level may be new contributors, or they may only contribute occasionally.
+
+### Maintainer
+
+Maintainers are established contributors who are responsible for the entire project. As such, they have the ability to approve PRs against any area of the project, and are expected to participate in making decisions about the strategy and priorities of the project.
+
+## Ways to Contribute
+
+We welcome many different types of contributions including:
+
+* New features
+* Builds, CI/CD
+* Bug fixes
+* Documentation
+* Issue Triage
+* Communications / Social Media / Blog Posts
+* Release management
+
+## Find an Issue
+
+We have good first issues for new contributors and help wanted issues suitable
+for any contributor. [good first issue](TODO) has extra information to
```

Comment:
Can we fix the "good first issue" and "help wanted" links? Nit: also might want to sentence case them (so like "Good first issue"), just because they're right after a period. I think the casing is probably off in other repos too though so this isn't a big deal